bundles/org.openhab.binding.govee/src/test/java/org/openhab/binding/govee/internal/GoveeSerializeGoveeHandlerTest.java

```diff
@@ -0,0 +1,157 @@
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandlerCallback;
+import org.openhab.core.types.State;
+
+import com.google.gson.Gson;
+
+/**
+ * @author Stefan Höhn - Initial contribution
+ */
+@NonNullByDefault
+public class GoveeSerializeGoveeHandlerTest {
+
+    private static final Gson GSON = new Gson();
+    private final String invalidValueJsonString = "{\"msg\": {\"cmd\": \"devStatus\", \"data\": {\"onOff\": 0, \"brightness\": 100, \"color\": {\"r\": 1, \"g\": 10, \"b\": 0}, \"colorTemInKelvin\": 9070}}}";
+
+    private static final Configuration CONFIG = createConfig(true);
+    private static final Configuration BAD_CONFIG = createConfig(false);
+
+    private static Configuration createConfig(boolean returnValid) {
+        final Configuration config = new Configuration();
+        if (returnValid) {
+            config.put("hostname", "1.2.3.4");
+        }
+        return config;
+    }
+
+    private static Thing mockThing(boolean withConfiguration) {
+        final Thing thing = mock(Thing.class);
+        when(thing.getUID()).thenReturn(new ThingUID(GoveeBindingConstants.THING_TYPE_LIGHT, "govee-test-thing"));
+        when(thing.getConfiguration()).thenReturn(withConfiguration ? CONFIG : BAD_CONFIG);
+
+        final List<Channel> channelList = Arrays.asList(
+                mockChannel(thing.getUID(), GoveeBindingConstants.CHANNEL_COLOR), //
+                mockChannel(thing.getUID(), GoveeBindingConstants.CHANNEL_COLOR_TEMPERATURE), //
+                mockChannel(thing.getUID(), GoveeBindingConstants.CHANNEL_COLOR_TEMPERATURE_ABS));
+
+        when(thing.getChannels()).thenReturn(channelList);
+        return thing;
+    }
+
+    private static Channel mockChannel(final ThingUID thingId, final String channelId) {
+        final Channel channel = Mockito.mock(Channel.class);
+        when(channel.getUID()).thenReturn(new ChannelUID(thingId, channelId));
+        return channel;
+    }
+
+    private static GoveeHandler createAndInitHandler(final ThingHandlerCallback callback, final Thing thing) {
+        final ScheduledExecutorService executorStub = Mockito.mock(ScheduledExecutorService.class);
+        doAnswer((InvocationOnMock invocation) -> {
+            ((Runnable) invocation.getArguments()[0]).run();
+            return null;
+        }).when(executorStub).scheduleWithFixedDelay(any(Runnable.class), anyLong(), anyLong(), any(TimeUnit.class));
+
+        final GoveeHandler handler = spy(new GoveeHandler(thing, new CommunicationManager(), executorStub));
+
+        handler.setCallback(callback);
+        handler.initialize();
+        return handler;
+    }
+
+    private static State getState(final int input) {
```

Comment:
This is unused, can you check the SAT report?

Reply:
Was intentionally, but ok. I'll also fix the other warnings not related to this PR.

Reply:
Maybe that would be better to do in a separate PR in case we should backport this one, WDYT?

Reply:
You are right, i'll leave those out to reduce the risk.